### PR TITLE
test: add missing assertions and edge cases for route-ids-for-agency

### LINKED ISSUE #971: spec-gap: Add missing test assertions and edge cases for route-ids-for-agency
### Description
The `route-ids-for-agency` endpoint implementation is solid, but the test suite is missing coverage for several specific requirements detailed in the specification. We need to expand the test coverage to ensure full spec compliance and prevent future regressions.

### Missing Coverage (Spec Gaps)
1. **Extension 4b (Zero Routes):** There is no test verifying that an agency existing in the database but having zero routes returns an HTTP 200 with an empty list `[]`.
2. **limitExceeded:** The success tests do not verify that `limitExceeded` is strictly `false`.
3. **References Block:** The tests only verify that `references.agencies` is empty. The spec requires asserting that **all six** sub-lists (`agencies`, `routes`, `stops`, `trips`, `situations`, `stopTimes`) are empty.
4. **Envelope Metadata:** Success tests lack assertions for `version == 2` and `currentTime > 0`.

### PR DESCRIPTION
**Pull request by 3rabiii**
## Description
This PR expands the test coverage for the `route-ids-for-agency` endpoint to ensure full compliance with the OneBusAway specification and prevent future regressions. 

### Changes Included:
1. **Zero Routes Edge Case (Extension 4b):** Added a new test case to verify that an agency existing in the database but having zero routes correctly returns an `HTTP 200 OK` with an empty `data.list` (`[]`).
2. **`limitExceeded` Assertion:** Updated existing success tests to explicitly assert that `limitExceeded` is strictly `false`.
3. **Comprehensive References Check:** Enhanced the test assertions to verify that all six sub-lists within the `references` block (`agencies`, `routes`, `stops`, `trips`, `situations`, `stopTimes`) are empty.
4. **Envelope Metadata Verification:** Added checks to ensure the response envelope correctly contains `version == 2` and `currentTime > 0` on successful requests.

fixes: #971 

## Summary by CodeRabbit

* **New Features**
  * Added the ability to delete individual agencies.

* **Tests**
  * Enhanced endpoint tests with additional response validations: version, timestamp, limit flag, and empty reference collections.
  * Added test coverage for agencies that have no associated routes.

### CODE REVIEW

File: gtfsdb/query.sql [hunk 1]
```diff
@@ -612,6 +612,9 @@ DELETE FROM routes;
 -- name: ClearAgencies :exec
 DELETE FROM agencies;
 
+-- name: DeleteAgency :exec
+DELETE FROM agencies WHERE id = ?;
```
Comment (burma-shave) [inline gtfsdb/query.sql]: Other tests define sql queries in-line using `DB.ExecContext`. We should follow the same pattern here.

**Review by burma-shave — CHANGES_REQUESTED** [review]

**Review by burma-shave — APPROVED** [review]
